### PR TITLE
RavenDB-16719: Dispose compare exchange values on import

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

File: src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs [hunk 1]
```diff
@@ -761,10 +761,12 @@ private SmugglerProgressBase.Counts ProcessCompareExchange(SmugglerResult result
             using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out JsonOperationContext context))
             using (var actions = _destination.CompareExchange(context))
             {
-                foreach (var kvp in _source.GetCompareExchangeValues())
+                foreach (var kvp in _source.GetCompareExchangeValues(actions))
                 {
```
Comment (grisha-kotler) [inline src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs:765]: Need to dispose the `StorageKey` as well

**Review by grisha-kotler — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Smuggler/Documents/Data/ISmugglerDestination.cs [hunk 2]
```diff
@@ -39,6 +39,11 @@ public interface IDocumentActions : INewDocumentActions, IDisposable
         IEnumerable<DocumentItem> GetDocumentsWithDuplicateCollection();
     }
 
+    public interface INewCompareExchangeActions
+    {
+        JsonOperationContext GetContextCompareExchange();
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/Data/ISmugglerDestination.cs]: GetContextForNewCompareExchangeValue

File: src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs [hunk 3]
```diff
@@ -761,10 +761,12 @@ private SmugglerProgressBase.Counts ProcessCompareExchange(SmugglerResult result
...
                     _token.ThrowIfCancellationRequested();
                     result.CompareExchange.ReadCount++;
+                    if (result.CompareExchange.ReadCount != 0 && result.CompareExchange.ReadCount % 1000 == 0)
+                        AddInfoToSmugglerResult(result, $"Read {result.CompareExchange.ReadCount:#,#;;0} compare exchanges.");
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs]: compare exchange values.

File: src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs (hunk 1 above)
Comment (garayx) [reply src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs:765]: the key is string

File: src/Raven.Server/Smuggler/Documents/StreamDestination.cs [hunk 4]
```diff
@@ -919,6 +919,11 @@ public void WriteTombstoneKey(string key)
                 Writer.WriteString(key);
                 Writer.WriteEndObject();
             }
+
+            public JsonOperationContext GetContextForNewCompareExchangeValue()
+            {
+                throw new NotSupportedException();
```
Comment (arekpalinski) [inline src/Raven.Server/Smuggler/Documents/StreamDestination.cs:925]: Is it okay to throw here?

Comment (garayx) [reply src/Raven.Server/Smuggler/Documents/StreamDestination.cs:925]: yes, this method never called, its needed to implement the interface

**Pull request merged**